src/ray/util/process.cc

```diff
@@ -326,26 +326,31 @@ pid_t ProcessFD::GetId() const { return pid_; }
 
 Process::~Process() {}
 
-Process::Process() {}
+Process::Process() : startup_token_(-1) {}
 
 Process::Process(const Process &) = default;
 
 Process::Process(Process &&) = default;
 
 Process &Process::operator=(Process other) {
   p_ = std::move(other.p_);
+  startup_token_ = other.GetStartupToken();
   return *this;
 }
 
-Process::Process(pid_t pid) { p_ = std::make_shared<ProcessFD>(pid); }
+Process::Process(pid_t pid, StartupToken startup_token) {
+  p_ = std::make_shared<ProcessFD>(pid);
+  startup_token_ = startup_token;
+}
 
 Process::Process(const char *argv[], void *io_service, std::error_code &ec, bool decouple,
-                 const ProcessEnvironment &env) {
+                 const ProcessEnvironment &env, StartupToken startup_token) {
   (void)io_service;
   ProcessFD procfd = ProcessFD::spawnvpe(argv, ec, decouple, env);
   if (!ec) {
     p_ = std::make_shared<ProcessFD>(std::move(procfd));
   }
+  startup_token_ = (startup_token >= 0) ? startup_token: GetNewStartupToken();
```

Comment:
I think we can always generate StartupToken inside Process constructor. This way we can also make _global_token a private static member of Process class.

Reply:
Currently the `argv` is built with various uses of `worker_startup_token_counter_` in `worker_pool.cc`, so the token is needed before the constructor is called. For Java it is added as `-Dray.raylet.startup-token=`, for Python and CPP it is added as `--startup-token=`.

Reply:
Oh, got it.